src/HDF5.jl

```diff
@@ -1411,16 +1413,16 @@ function read_row(io::IO, membertype, membersize)
     row = Any[]
     for (dtype, dsize) in zip(membertype, membersize)
         if dtype === String
-            push!(row, unpad(read!(io, Vector{UInt8}(dsize)), H5T_STR_NULLPAD))
+            push!(row, unpad(read!(io, @compat Vector{UInt8}(uninitialized, dsize)), H5T_STR_NULLPAD))
         elseif dtype<:HDF5.FixedArray && eltype(dtype)<:HDF5BitsKind
-            val = read!(io, Vector{eltype(dtype)}(prod(size(dtype))))
+            val = read!(io, @compat Vector{eltype(dtype)}(uninitialized, prod(size(dtype))))
             push!(row, reshape(val, size(dtype)))
         elseif dtype<:HDF5BitsKind
             push!(row, read(io, dtype))
         else
             # for other types, just store the raw bytes and let the user
             # decide what to do
-            push!(row, read!(io, Vector{UInt8}(dsize)))
+            push!(row, read!(io, @compat Vector{UInt8}(uninitialized, dsize)))
```

Comment:
I don't think any of the modifications in this file require `@compact`, otherwise lgtm